src/map/skill.cpp

```diff
@@ -9961,7 +9961,7 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 	case SR_ASSIMILATEPOWER:
 		if (flag&1) {
 			i = 0;
-			if (dstsd && dstsd->spiritball && (sd == dstsd || map_flag_vs(src->m)) && (dstsd->class_&MAPID_BASEMASK)!=MAPID_GUNSLINGER) {
+			if (dstsd && (dstsd->spiritball || dstsd->spiritcharm) && (sd == dstsd || map_flag_vs(src->m)) && (dstsd->class_&MAPID_BASEMASK)!=MAPID_GUNSLINGER) {
```

Comment:
Maybe it'd be better to just remove the `dstsd->spiritball && dstsd->spiritcharm` check here since we check again after this if-statement succeeds.

Reply:
Indeed! Took the oportunity to do the same on MO_ABSORBSPIRITS as the absorption of ninja charms was also unreachable.